src/covariances.jl

```diff
@@ -0,0 +1,80 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+using Statistics
+
+
+"""
+    designmatrix(comps)
+    
+Converts a vector of Composition objects into a Matrix of numbers.
+"""
+function designmatrix(comps)
+    matrix = reduce(hcat, components.(comps)[:,1])'
+    return matrix
+end
+
+"""
+    compvarmatrix(comps)
+
+Returns the compositional variation array, from definition 4.3 of Aitchson - The Statistical Analysis of Compositional Data.
+"""
+function compvarmatrix(comps)
+    dm = designmatrix(comps)
+    N, D = size(dm)
+    cva = zeros(D, D)
+
+    for i in 1:D
+        for j in i+1:D
+            log_ratios = log.(dm[:, i] ./ dm[:, j])
+            cva[j, i] = sum(log_ratios) / N
+            cva[i, j] = sum((log_ratios .- cva[j, i]).^2 ) / N
+        end
+    end
+
+    return cva
+end
+
+"""
+    variationmatrix(comps)
+
+Return the variation matrix, definition 4.4 of Aitchson - The Statistical Analysis of Compositional Data.
+"""
+function variationmatrix(comps)
+    dm = designmatrix(comps)
+    N, D = size(dm)
+    va = zeros(D, D)
```

Comment:
Let's use the corresponding notation of the book. In this case the matrix is denoted with T

Reply:
Done in commit aa2898e0c4dfdf5616f7d24604cfed53f0f56efc, but review is needed.